File: lcm.c
#include <stdio.h>

long long gcd(long long x, long long y)
{
    	if (x == 0) {
		return y;
	}
    	return gcd(y % x, x);
}

int main()
{
	int n;
	scanf("%d", &n);
	long long a[n], b[n], c[n];
	long long temp, temp2;
	for (int i = 0; i < n; i++) {
		scanf("%lld %lld", &a[i], &c[i]);
	}
	for (int i = 0; i < n; i++) {
		if ((a[i] > c[i]) || (c[i] % a[i] != 0)) {
			b[i] = -1;
			continue;
		}
		temp = c[i] / a[i];
		temp2 = gcd(a[i], temp);
		while (temp2 != 1) {
			temp = temp2;
			a[i] /= temp;
			temp2 = gcd(a[i], temp);
		}
		b[i] = c[i] / a[i];
	}
	for (int i = 0; i < n; i++) {
		printf("%lld\n", b[i]);
	}
	return 0;
}

File: lcm.py
import math

def gcd(x, y):
    if (x == 0):
        return y
    return gcd(y % x, x)

def find_b(a, c):
    b = list()
    for i in range(len(a)):
        if (a[i] > c[i] or c[i] % a[i] != 0):
            b.append(-1)
            continue
        temp = c[i] // a[i]
        temp2 = gcd(a[i], temp)
        while (temp2 != 1):
            temp = temp2
            a[i] //= temp
            temp2 = gcd(a[i], temp)
            
        b.append(c[i] // a[i])
        
    return b


if __name__ == '__main__':
    n = int(input())
    a = list()
    c = list()
    for i in range(n):
        temp1, temp2 = input().split()
        a.append(int(temp1))
        c.append(int(temp2))
    b = find_b(a, c)
    for number in b:
        print(number)
